app/searchers/responses_searcher.rb

```diff
@@ -91,6 +91,22 @@ def apply
       fragment.presence || "'00000000-0000-0000-0000-000000000000'"
     end
 
+    save_filter_data(search)
+
     relation.where(sql)
   end
+
+  # Save specific data that can be used for search filters.
+  def save_filter_data(search)
+    search.expressions.each do |expression|
+      if expression.qualifier.assoc.include?(:forms)
```

Comment:
Here I think you can just check if the name is `"form"`. The exact_form one is only used by reports and is kind of a hack.